x-pack/platform/plugins/shared/onechat/server/services/tools/esql/client.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { esqlToolProviderId } from '@kbn/onechat-common';
+import { logger } from 'elastic-apm-node';
+import { EsqlToolCreateRequest } from '../../../../common/tools';
+import { esqlToolIndexName } from './storage';
+import { EsqlToolStorage } from './storage';
+
+export interface EsqlToolClient {
+  get(toolId: string): Promise<EsqlToolCreateRequest>;
+  list(): Promise<EsqlToolCreateRequest[]>;
+  create(esqlTool: EsqlToolCreateRequest): void;
+  update(toolId: string, updates: Partial<EsqlToolCreateRequest>): Promise<EsqlToolCreateRequest>;
+  delete(toolId: string): Promise<boolean>;
+}
+
+export const createClient = ({ storage }: { storage: EsqlToolStorage }): EsqlToolClient => {
+  return new EsqlToolClientImpl({ storage });
+};
+
+class EsqlToolClientImpl {
+  public readonly id = esqlToolProviderId;
+  private readonly storage: EsqlToolStorage;
+
+  constructor({ storage }: { storage: EsqlToolStorage }) {
+    this.storage = storage;
+  }
+
+  async get(id: string): Promise<EsqlToolCreateRequest> {
+    try {
+      const document = await this.storage.getClient().get({ id });
+      const tool = document._source as EsqlToolCreateRequest;
+
+      return tool;
+    } catch (error) {
+      logger.error(`Error retrieving ESQL tool with ID ${id}: ${error}`);
+      throw error;
+    }
+  }
+
+  async list(): Promise<EsqlToolCreateRequest[]> {
+    try {
+      const document = await this.storage.getClient().search({
+        index: esqlToolIndexName,
+        query: {
+          match_all: {},
+        },
+        size: 1000,
+        track_total_hits: true,
+      });
+
+      return document.hits.hits.map((hit) => hit._source as EsqlToolCreateRequest);
+    } catch (error) {
+      logger.error(`Error fetching all ESQL tools: ${error}`);
+      throw error;
+    }
+  }
+
+  async create(tool: EsqlToolCreateRequest) {
+    try {
+      if (!tool.id) {
+        throw new Error('Tool ID is required');
+      }
+
+      const document = {
+        ...tool,
+        meta: {
+          providerId: esqlToolProviderId,
+          tags: [],
+        },
+      };
+
+      await this.storage.getClient().index({
+        id: tool.id,
+        document,
+      });
```

Comment:
Should we actually return what we indexed to indicate success? E.g. with `return this.get(id);` Like in conversation client https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/onechat/server/services/conversation/client.ts#L95